src/main/java/core/basesyntax/service/FruitTransaction.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.service;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int amount;
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getAmount() {
+        return amount;
+    }
+
+    public void setAmount(int amount) {
+        this.amount = amount;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation operationFromCode(String code) {
+            for (Operation operation : Operation.values()) {
+                if (operation.getCode().equals(code)) {
+                    return operation;
+                }
+            }
+            throw new RuntimeException("can't find operation");
```

Comment:
It's a good practice to throw exceptions with informative messages. Please provide more details in the message of the RuntimeException at line 54. For example, you can include the invalid code that caused the exception.